Serial ioctl() workaround

The ioctls.h of some architectures (notably xtensa) references structs from
linux/serial.h. Make sure to include this header as well.

Also, undef TIOCTTYGSTRUCT that require reference to internal kernel tty_struct,
but isn't actually referenced in modern kernels.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
diff -Nuar Python-2.7.8-orig/Modules/termios.c Python-2.7.8/Modules/termios.c
--- Python-2.7.8-orig/Modules/termios.c	2014-06-30 05:05:44.000000000 +0300
+++ Python-2.7.8/Modules/termios.c	2014-07-18 06:16:43.393010357 +0300
@@ -16,7 +16,9 @@
  * so this needs to be included first on that platform. */
 #include <termio.h>
 #endif
+#include <linux/serial.h>
 #include <sys/ioctl.h>
+#undef TIOCTTYGSTRUCT
 
 /* HP-UX requires that this be included to pick up MDCD, MCTS, MDSR,
  * MDTR, MRI, and MRTS (appearantly used internally by some things
